qiskit/visualization/gate_map.py

```diff
@@ -198,11 +208,72 @@ def plot_gate_map(backend, figsize=None,
         ax.add_artist(mpatches.Ellipse(
             _idx, width, height, color=qubit_color[var], zorder=1))
         if label_qubits:
-            ax.text(*_idx, s=str(var),
+            ax.text(*_idx, s=qubit_labels[var],
                     horizontalalignment='center',
                     verticalalignment='center',
                     color=font_color, size=font_size, weight='bold')
     ax.set_xlim([-1, x_max+1])
     ax.set_ylim([-(y_max+1), 1])
     plt.close(fig)
     return fig
+
+
+def plot_circuit_layout(circuit, backend, view='virtual'):
+    """Plot the layout of a circuit transpiled for a given
+    target backend.
+
+    Args:
+        circuit (QuantumCircuit): Input quantum circuit.
+        backend (BaseBackend): Target backend.
+        view (str): Layout view: either 'virtual' or 'physical'.
+
+    Returns:
+        Figure: A matplotlib figure showing layout.
+
+    Raises:
+        Exception: Circuit has no layout attribute.
+        ValueError: Invalid view type given.
+    """
+    if circuit.layout is None:
+        raise Exception('Circuit has no layout. '
```

Comment:
We probably should use a different exception from the base Exception here. I'd probably either use AttributeError or QiskitError.

Reply:
And done.